exchange/exchange_test.go

```diff
@@ -0,0 +1,415 @@
+
+	c := new(mockAdapter)
+	c.errs = make([]error, 0, 5)
+
+	e.adapterMap = map[openrtb_ext.BidderName]adaptedBidder{
+		BidderDummy: a,
+		BidderDummy2: b,
+		BidderDummy3: c,
+	}
+
+	e.adapters = make([]openrtb_ext.BidderName, 0, len(e.adapterMap))
+	for a, _ := range e.adapterMap {
+		e.adapters = append(e.adapters, a)
+	}
+	return e
+}
+
+
+func mockHandler(statusCode int, getBody string, postBody string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(statusCode)
+		if r.Method == "GET" {
+			w.Write([]byte(getBody))
+		} else {
+			w.Write([]byte(postBody))
+		}
+	})
+}
+
+func mockAdapterConfig1(a *mockAdapter) {
+	a.seatBid, a.errs = mockDummyBids1()
+
+	a.delay = 0 * time.Microsecond
+}
+
+func mockAdapterConfig2(a *mockAdapter) {
+	a.seatBid, a.errs = mockDummyBids2()
+
+	a.delay = 0 * time.Microsecond
+}
+
+func mockAdapterConfig3(a *mockAdapter) {
+	a.seatBid, a.errs = mockDummyBids3()
+
+	a.delay = 0 * time.Microsecond
+}
+
+func mockAdapterConfigErr1(a *mockAdapter) {
+	a.seatBid, a.errs = mockDummyBidsErr1()
+
+	a.delay = 0 * time.Microsecond
+}
+
+func mockAdapterConfigErr2(a *mockAdapter) {
+	a.seatBid, a.errs = mockDummyBidsErr2()
+
+	a.delay = 0 * time.Microsecond
+}
+
+func mockDummyBids1() (*pbsOrtbSeatBid, []error) {
+	sb1 := new(pbsOrtbSeatBid)
+	sb1.bids = make([]*pbsOrtbBid, 2)
+	sb1.bids[0] = new(pbsOrtbBid)
+	sb1.bids[1] = new(pbsOrtbBid)
+	sb1.bids[0].bid = new(openrtb.Bid)
+	sb1.bids[1].bid = new(openrtb.Bid)
+	sb1.bids[0].bid.ID = "1234567890"
+	sb1.bids[1].bid.ID = "5678901234"
+
+	errs := make([]error, 0, 5)
+
+	return sb1, errs
+}
+
+func mockDummyBids2() (*pbsOrtbSeatBid, []error) {
+	sb1 := new(pbsOrtbSeatBid)
+	sb1.bids = make([]*pbsOrtbBid, 2)
+	sb1.bids[0] = new(pbsOrtbBid)
+	sb1.bids[1] = new(pbsOrtbBid)
+	sb1.bids[0].bid = new(openrtb.Bid)
+	sb1.bids[1].bid = new(openrtb.Bid)
+	sb1.bids[0].bid.ID = "ABC"
+	sb1.bids[1].bid.ID = "1234"
+
+	errs := make([]error, 0, 5)
+
+	return sb1, errs
+}
+func mockDummyBids3() (*pbsOrtbSeatBid, []error) {
+	sb1 := new(pbsOrtbSeatBid)
+	sb1.bids = make([]*pbsOrtbBid, 1)
+	sb1.bids[0] = new(pbsOrtbBid)
+	sb1.bids[0].bid = new(openrtb.Bid)
+	sb1.bids[0].bid.ID = "MyBid"
+
+	errs := make([]error, 0, 5)
+
+	return sb1, errs
+}
+func mockDummyBidsErr1() (*pbsOrtbSeatBid, []error) {
+	sb1 := new(pbsOrtbSeatBid)
+	sb1.bids = nil
+
+	errs := make([]error, 0, 5)
+	errs = append(errs, errors.New("This was an error") )
+	errs = append(errs, errors.New("Another error goes here") )
+
+	return sb1, errs
+}
+func mockDummyBidsErr2() (*pbsOrtbSeatBid, []error) {
+	var sb1 *pbsOrtbSeatBid = nil
+
+	errs := make([]error, 0, 5)
+	errs = append(errs, errors.New("This was a FATAL error") )
+
+	return sb1, errs
+}
+
+
+func convertErr2Str(e []error) []string {
+	s := make([]string, len(e))
+	for i :=0; i<len(e); i++ {
+		s[i] = e[i].Error()
+	}
+	return s
+}
+
+// errorString is a trivial implementation of error.
+type errorString struct {
```

Comment:
Is this dead code? I don't see it used, and It looks like `errors.New(str)` basically does this anyway.

Reply:
Probably is dead, can certainly be removed in favor of `fmt.Error()` or `errors.New()` if it is still used.